tests/unit/orchestrate/flow/flow-construct/test_flow_except.py

```diff
@@ -263,3 +263,23 @@ def test_flow_does_not_import_exec_depencies():
     with pytest.raises(RuntimeFailToStart):
         with f:
             pass
+
+
+def test_flow_head_runtime_failure(monkeypatch, capfd):
+    from jina.serve.runtimes.request_handlers.data_request_handler import (
+        DataRequestHandler,
+    )
+
+    def fail(*args, **kwargs):
+        raise NotImplementedError('Intentional error')
+
+    monkeypatch.setattr(DataRequestHandler, 'merge_routes', fail)
+
+    with Flow().add(shards=2) as f:
+        f.index(
+            [Document(text='abbcs')],
+        )
+
+    out, err = capfd.readouterr()
```

Comment:
what is this capfd?

Reply:
It's a pytest fixture that captures the std and err outputs, so it lets us check that the correct error message is shown to the user